src/python/pants/engine/rules.py

```diff
@@ -41,12 +44,36 @@
 PANTS_RULES_MODULE_KEY = "__pants_rules__"
 
 
+def implicitly(*args) -> dict[str, Any]:
```

Comment:
This needs all of the same docs that currently exist on `Get` regarding syntax/usage. But I figure that that can wait until we're ready to actually recommend this syntax.